### PR TITLE
fix: added "null" check when checking for duplicate keys in case of autoid. closes #2822

### LINKED ISSUE #2822: Uploading new data using Excel file fails in case of auto_id
**What version of EMX2 are you using (see footer)**
v8.221.3

**Describe the bug**
When I upload multiple new rows via Excel and the sheet contains the new data without the primary key = auto_id column the upload fails with a Duplicate key error:
image

- It works in case of only one new row in your Excel file.
- Uploading the new rows without the auto_id column saved in a csv file works fine 

**To Reproduce**
1. Go to 'https://emx2.dev.molgenis.org'
2. Click on pet_store

3. Upload/Download 
4. Import the attached Excel file 
[test_new_pet_order.xlsx](https://github.com/molgenis/molgenis-emx2/files/12802227/test_new_pet_order.xlsx)
5. See the error

**Expected behavior**
I expect to be able to upload multiple rows using an Excel file in case the primary key = auto_id column is not included in the Excel file. 

**Screenshots**

**Desktop (please complete the following information):**
 - iOS
 - Safari

### PR DESCRIPTION
**Pull request by harmbrugge**
The ValidatePkeyProcessor fails when checking primary keys with empty values:

Collectors.joining(",") returns an "null" string when the primary key value is absent, therefore "null" is added to the keys. Therefore if multiple rows are checked with empty pkey values, they will be incorrectly flagged as duplicates. 

Not sure if this is the right place to fix it

fixes #2822

### CODE REVIEW

File: backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/ImportTableTask.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public void process(Iterator<Row> iterator, TableStore source) {
                   .map(Field::getName)
                   .collect(Collectors.joining(","));
           task.addSubTask("Found duplicate Key (" + keyFields + ")=(" + keyValue + ")").setError();
-        } else {
+        } else if (!keyValue.equals("null")) {
```
Comment (chinook25) [inline backend/molgenis-emx2-io/src/main/java/org/molgenis/emx2/io/ImportTableTask.java:107]: Is it a good idea to instead check for if(keyValue.equals("null") and throw an error in that case? (just an idea, not sure if that's what you'd want here)

**Comment by harmbrugge** [comment]
Now I have added a check for empty primary keys when the ColumnType is not AUTO_ID. I add the the errors to the task and check at completion if errors were present.

**Comment by harmbrugge** [comment]
Added the test

**Comment by mswertz** [comment]
@harmbrugge any idea why build fails?

**Comment by harmbrugge** [comment]
> @harmbrugge any idea why build fails?
If a primary key is a reference and a compound key, the reference values were not added to the duplication check.
Fixed it by gathering all the columns at object creation.

**Review by mswertz — APPROVED** [review]

**Review by dtroelofsprins — APPROVED** [review]
Only did a functional review:
- Upload of file without the ID column into a table with autoID works fine now!
- Upload of a file without an ID column into a table with no autoID column still fails (which should also be the case).

**Pull request merged**